src/routes/MenuDetailPage.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { ShoppingCartOutlined } from "@ant-design/icons";
+import {Badge, Flex, Layout, Select} from "antd";
+import MenuItemDetailCard from "../components/MenuItemDetailCard";
+import Heading from "../components/Heading";
+import {CustomInput} from "../components/CustomInputs";
+import CustomButton from "../components/CustomButton";
+import HeaderNav from "../components/HeaderNav";
+import {useLocation, useNavigate} from "react-router-dom";
+const { Content} = Layout;
+
+const MenuDetailPage = () => {
+    const navigate = useNavigate();
+    const {state} = useLocation();
+    const allOrders = state.allOrders;
+    const userOrder = state.userOrder;
+    const menuItem = state.menuItem;
+    const pendingOrder = {item: menuItem}
+
+
+    const initPortionOptions = () => {
+        let portionOptions = []
+        for (let size in menuItem.portionOptions) {
+            let portionLabel = size.toUpperCase() + " - $" + menuItem.portionOptions[size]
+            portionOptions.push({value: size, label: portionLabel})
+        }
+        return portionOptions
+    }
+
+    const onClickCart = () => {
+        navigate('/user-home/order-summary', {state: {
+                order: userOrder,
+                allOrders: allOrders
+        }})
+    }
+
+    const onSelectionChange = (value) => {
+        pendingOrder["size"] = value
+    }
+
+    const onAmountChange = (e) => {
+        pendingOrder["count"] = Number(e.target.value);
+    }
+
+    const onAddToOrderClick = () => {
+        let pricePerItem = menuItem.portionOptions[pendingOrder.size]
+        //TODO: sanity check potion selection & amount
+        userOrder.orderItems.push(pendingOrder)
+        userOrder.totalPrice += pendingOrder.count * pricePerItem;
+        userOrder.totalCount += pendingOrder.count
+
+        navigate('/user-home', {state: {userOrder: userOrder, allOrders: allOrders}})
+    }
+
+    const onBackButtonClick = () => {
+        navigate('/user-home', {state: {userOrder: userOrder, allOrders: allOrders}})
```

Comment:
I'm afraid this discussion https://github.com/congruiyS2023/platemate-web/pull/17#discussion_r1416338622 is not resolved. See the demo video below. One place I observed that caused this issue is here. The `fromJoinSuccess` state is not passed here when you click the back button at top nav bar when you're in the item detail page. One workaround I could see here would be just put `joinSuccess` to `localStorage`, and check it across page navigation. What do you think? @xyang6230 

https://github.com/congruiyS2023/platemate-web/assets/120247054/2b283254-945c-4680-b701-42a8072a2920

Reply:
I see, I only fix the pop up modal but not aware of this string. Let me add fromJoinSuccess state

Reply:
Cool.